README.md

```diff
@@ -347,6 +348,53 @@ let g:ycm_language_server =
 Place crystalline in the path (i.e. /usr/local/bin) or use absolute path
 in the example above..
 
+# Astro
+
+In addition to defining `g:ycm_language_server` block as shown in
+[`astro/snippet.vim`](astro/snippet.vim) this LSP requires `.ycm_extra_conf.py`
+to pass Language Server `initializationOptions` pointing to a directory
+containing either `typescript.js` or `tsserverlibrary.js` file, such as
+`node_modules/typescript/lib`, via `typescript.tsdk` key/value address, eg.
+
+```python
+import os
+
+def Settings( **kwargs ):
+    current_directory = os.path.abspath(os.path.curdir)
+    if kwargs[ 'language' ] == 'astro':
+        configs = {
+            'ls': {
+                'typescript': {
+                    'tsdk':  f"{current_directory}/node_modules/typescript/lib"
+                },
+            },
+        }
+
+        return configs
+```
+
+...  Authors of Astro Language Server recommend installing `prettier` plugins
+too, so adding the following to your `npm init` rituals may be a good idea;
+
+```bash
+npm install --save-dev typescript prettier prettier-plugin-astro @astrojs/ts-plugin
```

Comment:
Just one more question, because I am not too familiar with the node ecosystem:
Would things work as expected if we were to put these four things into our `astro/package.json`?
Especially `typescript` would be useful as then we would know exactly where it is located.
Am I overlooking anything?

Reply:
Though that'd work in many cases, almost most, we'd run into issues where a project's version and what's installed within `astro/node_modules` would introduce misbehavior _opportunities_.

> Note; those concerned about space constraints could use [`pnpm`](https://github.com/pnpm/pnpm), or [`NODE_PATH`](https://nodejs.org/api/modules.html#loading-from-the-global-folders), and accept the added _fun_ when stuff breaks.
>
> One way to think about Node/NPM ecosystem is `npm install` behavior is almost analogous to Python's `venv` stuff, and `pnpm` (when it works) is almost as nice as `pipx`

However, I'd advise keeping it to a per-project setup ceremony not just because of previous reasons, but also because project contributors will likely need to install the same/similar dependencies.

Reply:
Ok, let's not do that.